### DISCUSSION TITLE
Strange problem with the selection buttons

### DISCUSSION THREAD

**Discussion by TBDwarf** [searxng/searxng#2821] [General]
Hello,

I don't know where to turn, so I'm going to ask the community via the discussion tab.
For some time now (I'd say about a week), I've been experiencing a strange phenomenon on my Searxng instance.

My search type selection boxes are displayed abnormally, at first I wasn't worried, I thought the problem was probably just a bug that would be resolved in some update, after a week I went to look at other instances and I realised that I'm the only one suffering from this display anomaly (see image).
![image](https://github.com/searxng/searxng/assets/79540750/8cbfb3a9-ae75-4e80-bf9e-da677383ac3d)

I'm running Searxng under Docker, I've completely deleted then recreated my container without success.
Does anyone have any idea how to solve my problem? I can't find any conversation on this subject...

Thanks in advance.

**Comment by return42** [comment]
- Is you instance public / where we can reproduce the issue?
- Did you customized any CSS?

Clear browser's cache .. see [label:"clear browser cache"](https://github.com/search?q=repo%3Asearxng%2Fsearxng+label%3A%22clear+browser+cache%22)

**Reply by TBDwarf** [reply]
Thanks for trying to help me, this instance is private but accessible directly from the internet (if needed I can give you the link). I didn't do any CSS customization I only touched the setting file (to disable some search engine and define a default language).
Where I'm surprised, I deleted my Docker searxng container (running on a Debian VPS), I deleted the images etc.. I recreated my container from 0 (I only kept the setting file) and the problem remains the same...

The problem appeared spontaneously overnight (I have a daily container purge with a latest pull).

**Reply by dalf** [reply]
Have you tried to clear the browser cache?
The CSS has changed few days ago : https://github.com/searxng/searxng/pull/2740

**Reply by TBDwarf** [reply]
I feel like an idiot, for me I had cleared this hypothesis by blocking the browser cache with the Firefox inspection tool (and the problem was present), I've just opened a private browsing window without conviction and indeed it was the cache that was the source of my problem! I'm sorry for the lost time, thanks again for the help, you're great. I hope my case will be useful to others!
I haven't tested it on my PC browser (only on my phone) but there's no reason for it to be anything else.

**Reply by dalf** [reply]
You can set this value to `true` in settings.yml :
https://github.com/searxng/searxng/blob/d013f51a2588fbe9bee05cd0e3f46c75bf785460/searx/settings.yml#L104

However you have to configure Apache / nginx / whatever to deal with the `?<hash of the content>`. It may work with URL rewrite, I'm not sure, or you can configure uwsgi to serve the static file as it is done in the Docker image : https://github.com/searxng/searxng/blob/d013f51a2588fbe9bee05cd0e3f46c75bf785460/dockerfiles/uwsgi.ini#L41-L46

See 
* https://github.com/searxng/searxng/pull/932#issuecomment-1067039518
* https://github.com/searxng/searxng/pull/995

As described, it becomes easily complicated to deal with the query parameter and the reverse proxy, but one of the above suggestions might work for you.

**Reply by saukkico** [reply]
I had exactly the same problem. Running Searxng on Docker and accesing it through Cloudflare tunnel. I tried to use private browsing window, but didn't help.  I tried to access the instance by the server LAN ip address and then the problem went away. So for some reason the problem only occurs when accessing the instance through Cloudflare tunnel.
Changing the static_use_hash to `true` solved the problem for me.
